src/ToasterServiceProvider.php

```diff
@@ -25,10 +25,9 @@ public function boot(): void
             $this->registerPublishing();
         }
 
-        $this->relayToSession();
-
         $this->callAfterResolving(BladeCompiler::class, $this->aliasToasterHub(...));
         $this->callAfterResolving(Collector::class, $this->relayToLivewire(...));
+        $this->callAfterResolving(Router::class, $this->relayToSession(...));
```

Comment:
BTW, you may use booted function, to boot your dependencies. It is a little bit easier to read since order doesn't matter. You just need to register after all base components are booted.
```
$this->booted(function() {
     $this->aliasToasterHub();
     $this->relayToLivewire();
     $this->relayToSession();
});

```

Reply:
Hey, thanks for the suggestion but I'm going to pass on this because the reason I am using `callAfterResolving` is to register the services **only if the component is being used**. Consequently, it allows us to prevent some unnecessary registration on certain requests.

The way you suggest means that I'll have to eagerly register them instead of lazily registering them.